account_move_name_sequence/tests/test_account_move_name_seq.py

```diff
@@ -0,0 +1,136 @@
+            {
+                "date": self.date,
+                "journal_id": self.misc_journal.id,
+                "line_ids": [
+                    (0, 0, {"account_id": self.account1.id, "debit": 10}),
+                    (0, 0, {"account_id": self.account2.id, "credit": 10}),
+                ],
+            }
+        )
+        self.assertEqual(move.name, "/")
+        move.action_post()
+        seq = self.misc_journal.sequence_id
+        move_name = "%s%s" % (seq.prefix, "1".zfill(seq.padding))
+        move_name = move_name.replace("%(range_year)s", str(self.date.year))
+        self.assertEqual(move.name, move_name)
+        self.assertTrue(seq.date_range_ids)
+        drange_count = self.env["ir.sequence.date_range"].search_count(
+            [
+                ("sequence_id", "=", seq.id),
+                ("date_from", "=", fields.Date.add(self.date, month=1, day=1)),
+            ]
+        )
+        self.assertEqual(drange_count, 1)
+        move.button_draft()
+        move.action_post()
+        self.assertEqual(move.name, move_name)
+
+    def test_in_refund(self):
+        in_refund_invoice = self.env["account.move"].create(
+            {
+                "journal_id": self.purchase_journal.id,
+                "invoice_date": self.date,
+                "partner_id": self.env.ref("base.res_partner_3").id,
+                "move_type": "in_refund",
+                "invoice_line_ids": [
+                    (
+                        0,
+                        0,
+                        {
+                            "account_id": self.account1.id,
+                            "price_unit": 42.0,
+                            "quantity": 12,
+                        },
+                    )
+                ],
+            }
+        )
+        self.assertEqual(in_refund_invoice.name, "/")
+        in_refund_invoice.action_post()
+        seq = self.purchase_journal.refund_sequence_id
+        move_name = "%s%s" % (seq.prefix, "1".zfill(seq.padding))
+        move_name = move_name.replace("%(range_year)s", str(self.date.year))
+        self.assertEqual(in_refund_invoice.name, move_name)
+        in_refund_invoice.button_draft()
+        in_refund_invoice.action_post()
+        self.assertEqual(in_refund_invoice.name, move_name)
+
+    def test_remove_invoice_error(self):
+        invoice = self.env["account.move"].create(
+            {
+                "date": self.date,
+                "journal_id": self.misc_journal.id,
+                "line_ids": [
+                    (0, 0, {"account_id": self.account1.id, "debit": 10}),
+                    (0, 0, {"account_id": self.account2.id, "credit": 10}),
+                ],
+            }
+        )
+        self.assertEqual(invoice.name, "/")
+        invoice.action_post()
+        with self.assertRaises(UserError):
+            invoice.unlink()
+        invoice.button_draft()
+        invoice.button_cancel()
+        with self.assertRaises(UserError):
```

Comment:
You should use `assertRaisesRegex` instead whenever possible, to ensure the raised error is the expected one, e.g.:

```
error_msg = "could not perform the this operation because..."
with self.assertRaisesRegex(UserError, error_msg):
    ...
```

Same below.